curp/src/client/unary.rs

```diff
@@ -55,75 +51,80 @@ impl std::fmt::Debug for State {
 }
 
 /// Unary builder
-pub(super) struct UnaryBuilder<P: Protocol> {
+pub(super) struct UnaryBuilder {
     /// All members (required)
     all_members: HashMap<ServerId, Vec<String>>,
     /// Unary config (required)
     config: UnaryConfig,
-    /// Local server (optional)
-    local_server: Option<(ServerId, P)>,
-    /// Leader id (optional)
-    leader: Option<ServerId>,
-    /// Term (optional)
-    term: Option<u64>,
+    /// Leader state (optional)
+    leader_state: Option<(ServerId, u64)>,
     /// Cluster version (optional)
     cluster_version: Option<u64>,
 }
 
-impl<P: Protocol> UnaryBuilder<P> {
+impl UnaryBuilder {
     /// Create unary builder
     pub(super) fn new(all_members: HashMap<ServerId, Vec<String>>, config: UnaryConfig) -> Self {
         Self {
             all_members,
             config,
-            local_server: None,
-            leader: None,
-            term: None,
+            leader_state: None,
             cluster_version: None,
         }
```

Comment:
Same as above

Reply:
Implementing Default for UnaryConfig is not very appropriate; its default value comes from the utils crate.